publish.go

```diff
@@ -164,3 +168,36 @@ type BatchItemsError []BatchItemResponse
 func (err BatchItemsError) Error() string {
 	return "one or many events may have not been published"
 }
+
+// Format implements fmt.Formatter for BatchItemsError
+func (err BatchItemsError) Format(s fmt.State, verb rune) {
+	switch verb {
+	case 'v':
+		if s.Flag('+') {
+			var messages []string
+			for k, v := range err {
+				messages = append(messages, fmt.Sprintf("[%d]: %+v", k, v))
+			}
+			sort.Strings(messages)
+
+			builder := bytes.NewBuffer(make([]byte, 0))
+			switch len(err) {
```

Comment:
I think we could simplify this if we dropped `case 1` and just wrote an error message to `s` and returned in case `len(err)` is 0. I think `case 1` is kind of useless and if we really needed something like this than we should probably add the count of errors into the message instead.

Reply:
this is done to have grammatically correct error message, where `case 1` should have `an error occurred` message rather than `errors occurred`